google/cloud/pubsub/internal/tracing_batch_callback_test.cc

```diff
@@ -165,6 +164,41 @@ TEST(TracingBatchCallback, SubscribeAttributesForOrderingKey) {
                                  "ordering-key-0")))));
 }
 
+TEST(TracingBatchCallback, SubscribeAttributesForExactlyOnce) {
+  auto span_catcher = InstallSpanCatcher();
+  auto mock = std::make_shared<pubsub_testing::MockBatchCallback>();
+  EXPECT_CALL(*mock, callback).Times(1);
+  auto batch_callback = MakeTestBatchCallback(std::move(mock));
+  std::shared_ptr<opentelemetry::context::propagation::TextMapPropagator>
+      propagator = std::make_shared<
+          opentelemetry::trace::propagation::HttpTraceContext>();
+  BatchCallback::StreamingPullResponse response;
+  google::pubsub::v1::StreamingPullResponse r;
```

Comment:
Nit:  I would delay instantiating these objects until just before you need them.

Reply:
Done